typings/index.d.ts

```diff
@@ -127,28 +131,34 @@ interface QueryByAPI {
 interface FindByAPI {
   findByText: (
     text: string | RegExp,
-    waitForOptions?: WaitForOptions
+    waitForOptions?: WaitForOptions,
+    options?: TextMatchOptions,
   ) => FindReturn;
   findByPlaceholderText: (
     placeholder: string | RegExp,
-    waitForOptions?: WaitForOptions
+    waitForOptions?: WaitForOptions,
+    options?: TextMatchOptions,
   ) => FindReturn;
   findByDisplayValue: (
     value: string | RegExp,
-    waitForOptions?: WaitForOptions
+    waitForOptions?: WaitForOptions,
+    options?: TextMatchOptions,
   ) => FindReturn;
   findByTestId: (testID: string | RegExp, waitForOptions?: WaitForOptions) => FindReturn;
   findAllByText: (
     text: string | RegExp,
-    waitForOptions?: WaitForOptions
+    waitForOptions?: WaitForOptions,
+    options?: TextMatchOptions,
```

Comment:
It seems that RTL uses different args order for all findBy queries: text, queryOptions, waitForOptions. I think we should use the same order.

Reply:
This was intentional decision, because switching arguments would be a breaking change. Maybe worth to have this feature in the next major? I doubt many use waitForOptions at all so maybe we could get away with this, but maybe not

Reply:
I generally thing that 1) findBy is least used query type of [getBy, queryBy, findBy] and 2) I think that matchOptions are more used than waitForOptions, as some users seem to prefer { exact: false} over /regex/ query.

Overall, I am for breaking change with minor version bump, and docs note. 

As a help for the users, we could emit error/warning if matchOptions contain fields from waitForOptions (timeout, interval) to ease the transition.

Reply:
It's kinda missing feature we should have ship in v7 anyway...

Reply:
I think I've got it, as a compat feature till next major release we would inspect `queryOptions` for `timeout` and `interval` props and use them if they exist, but issue a console warning asking user to migrate their code.

This won't break the older code, but will notify the user about API change. For us it means a relatively small change in next major release (vs alternative of re-ordering the `queryOptions` and `waitForOptions` args for all potential users).

WDYT?